etc/silk-create-asset-group.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# Check for required commands:
+if ! command -v jq > /dev/null 2>&1; then
+    echo "jq not found. Install jq"
+    exit 1
+fi
+
+if ! command -v curl > /dev/null 2>&1; then
+    echo "curl not found. Install curl"
+    exit 1
+fi
+
+# Check for required environment variables:
+: "${silk_client_id:?}"
+: "${silk_client_secret:?}"
+: "${branch:?}"
+
+# Get Silk token:
+json_payload=$(cat <<EOF
+{
+    "client_id": "${silk_client_id}",
+    "client_secret": "${silk_client_secret}"
+}
+EOF
+)
+silk_jwt_token=$(curl -s -X POST "https://silkapi.us1.app.silk.security/api/v1/authenticate" \
+  -H "Accept: application/json" \
+  -H "Content-Type: application/json" \
+  -d "$json_payload" \
+  | jq -r '.token')
```

Comment:
Something that I missed this during initial review but am discovering now as I perform the same changes in C driver: The `curl` commands may need to be modified for error-resilience:

- Replace `-s` with `--no-progress-meter` (allow Curl to speak when it encounters errors, but don't generate the progress output)
- Add `--fail` (Make Curl exit non-zero if the HTTP status indicates an error, otherwise a 40x/50x will proceed without error)
- Add `--location` (Make curl follow HTTP redirects)

Reply:
Thank you! Applied in https://github.com/mongodb/libmongocrypt/pull/814/files